### PR TITLE
Added POST request for /meals

### PR DESCRIPTION
**Pull request by DanielItkes**
# Added: 
* Logic for post request for /meals endpoint.
* A few methods in our main "data collection" class to get a specific data item.
* A method to generate IDs.

# Issues: 
* In the GET request for '/meals' endpoint we should return a JSON object indexed by ID, right now we are returning just the 'meals' dictionary.
* To calculate nutritional values we need to use the Ninja API thing, right now we are doing it ourselves.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -7,10 +7,40 @@
 
 class DishesCollection:
     def __init__(self):
+        self.meal_id_counter = 0
```
Comment (OmerGeva) [inline main.py:10]: this should be in a seperate Meal class, as the logic is different from dishes

File: main.py [hunk 2]
```diff
@@ -7,10 +7,40 @@
...
         # TODO: CHANGE TO DYNAMIC
-        self.dishes = [{ "id": 1, "name": 'lasagna', 'rating': 5}, { "id": 2, 'name': 'chicken parm', 'rating': 3}]  
+        self.dishes = [{ 'id': 1, 'name': 'lasagna', 'cal': 1, 'sodium': 1, 'sugar': 1},
+                       { 'id': 2, 'name': 'lasagna', 'cal': 1, 'sodium': 1, 'sugar': 1},
+                       { 'id': 3, 'name': 'lasagna', 'cal': 1, 'sodium': 1, 'sugar': 1}]  
+    
     def get_dishes(self):
         return self.dishes
+    
+    def get_meals(self):
```
Comment (OmerGeva) [inline main.py:22]: in separate meal class

File: main.py [hunk 3]
```diff
@@ -23,10 +53,76 @@ def get(self):
...
+# Meal classes
+class MealsList(Resource):
+    global col
+    
+    def get(self):
+        return col.get_meals(), 201
+    
+    def post(self):
+        # Only accept app/json content type
+        if request.content_type != 'application/json':
```
[the remark this answers is not in the document]
Comment (DanielItkes) [reply main.py:68]: image
Can it be done in another way?

File: main.py [hunk 4]
```diff
@@ -7,10 +7,40 @@
...
+                       { 'id': 2, 'name': 'lasagna', 'cal': 1, 'sodium': 1, 'sugar': 1},
+                       { 'id': 3, 'name': 'lasagna', 'cal': 1, 'sodium': 1, 'sugar': 1}]  
+    
     def get_dishes(self):
         return self.dishes
+    
+    def get_meals(self):
+        return self.meals
+    
+    def add_meal(self, meal):
```
[the remark this answers is not in the document]
Comment (DanielItkes) [reply main.py:26]: Switched to general db class.

File: main.py (hunk 2 above)
Comment (DanielItkes) [reply main.py:23]: Switched to general db class.

File: main.py (hunk 1 above)
Comment (DanielItkes) [reply main.py:11]: Switched to general db class.

File: main.py [hunk 5]
```diff
@@ -23,10 +53,76 @@ def get(self):
...
+            return -2, 422
+        
+        # Calculate the total nutrition of the dishes, returns error if a dish doesn't exist
+        if not self.calculate_nutrition(req_json):
+            return -6, 422
+            
+        # Add meal into database
+        req_json['id'] = col.get_id('meal')
+        col.add_meal(req_json)
+        return req_json['id'], 201
```
[the remark this answers is not in the document]
Comment (DanielItkes) [reply main.py:88]: image

File: main.py [hunk 6]
```diff
@@ -7,10 +7,40 @@
...
+    
+    def get_meals(self):
+        return self.meals
+    
+    def add_meal(self, meal):
+        self.meals.append(meal)
+    
+    # Generate ID for new datd items
+    def get_id(self, type):
+        if type == 'meal':
```
[the remark this answers is not in the document]
Comment (DanielItkes) [reply main.py:31]: Switched to general db class.

File: main.py [hunk 7]
```diff
@@ -23,10 +53,76 @@ def get(self):
...
+            return False
+        
+        # Check if values have the correct type
+        for key in json:
+            if type(json[key]) != required_params[key]:
+                return False
+        
+        return True
+    
+    def calculate_nutrition(self, json):
```
[the remark this answers is not in the document]
Comment (DanielItkes) [reply main.py:107]: Done.

File: main.py [hunk 8]
```diff
@@ -23,10 +53,76 @@ def get(self):
     def delete(self):
         return { }, 405
 
+# Meal classes
+class MealsList(Resource):
+    global col
+    
+    def get(self):
```
[the remark this answers is not in the document]
Comment (DanielItkes) [reply main.py:63]: Done.

File: main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (OmerGeva) [reply main.py:68]: let's keep this for now

**Review by OmerGeva — APPROVED** [review]

**Pull request merged**